doc/source/f2py/asterisk1.f90

```diff
@@ -0,0 +1,5 @@
+subroutine foo1(s)
+  character*(*), intent(out) :: s
+  !f2py character(f2py_len=12) s
+  s = "123456789A12"
```

Comment:
I'm wondering about this particular example. In this case, even though a compiler doesn't know what the length is; `f2py` has access to `s` and could perhaps infer this?

Of course this breaks down if this is something passed in at runtime.

Reply:
The Fortran compilers assume that `out` has enough space to write `"123456789A12"` and the caller is responsible for providing the memory for `s`. The `f2py_len` is introduced to instruct f2py to allocate the requested memory before entering the Fortran program. Otherwise, the behaviour is undefined and Fortran compiler dependent.

Reply:
Yup the directive and implementation makes a lot of sense, but I was wondering if it should be inferred by `f2py` without the explicit comment.

Reply:
This would not be possible. To f2py, the Fortran declaration
```
character*(*), intent(out) :: s
```
does not provide any information about the length of `s`, hence, we'll need to specify the length explicitly via f2py directive.

Recall f2py does not scan for Fortran execution statements, and even if it did, the required information may not be readily available anyway as is the case for the given example.